site-builder/src/main.rs

```diff
@@ -212,8 +212,50 @@ async fn run() -> Result<()> {
 
     let args = Args::parse();
     let mut config: Config = std::fs::read_to_string(&args.config)
+        .or_else(|err| {
```

Comment:
Hint: this code can be made much smaller. I would borrow from the walrus cli code directly as in here. https://github.com/MystenLabs/walrus/blob/c567eef5179776756ea7b0d4c0c930830695c41a/crates/walrus-service/src/common/utils.rs#L513-L524

Reply:
Done ✅